client/web/src/Structure/Admin/Components/QuestionsByTopic.jsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from 'react'
+import { adminActions } from '../State/action'
+import { useSelector, useDispatch } from "react-redux";
+import { Row } from '../'
+import Table from '@material-ui/core/Table';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import TableBody from '@material-ui/core/TableBody';
+import TablePagination from '@material-ui/core/TablePagination';
+
+export const QuestionsByTopic = ({val, handleDelete}) => {
+    const dispatch = useDispatch();
+    const {data} = useSelector( state => state.admin )
```

Comment:
don't use destructuring in useSelector, will lead to extra render. dot reference it.